pandas/core/window/rolling.py

```diff
@@ -318,7 +318,11 @@ def _prep_values(self, values: ArrayLike) -> np.ndarray:
             # GH #12373 : rolling functions error on float32 data
             # make sure the data is coerced to float64
             try:
-                values = ensure_float64(values)
+                if hasattr(values, "to_numpy"):
```

Comment:
we should change ensure_float64 to just do this directly (eg put your change there) - need to check perf and that this doesn't break anything

Reply:
can do isinstance instead of hasattr?